cse210-fa24-group3/cse210-fa24-group3
# Optimize Server Configuration and Improve Build Performance

# Pull Request Description

## Changes Made
- Added proper error handling in server.js
- Optimized static file serving with caching configuration
- Added environment variable support
- Simplified package.json by removing unnecessary dependencies
- Added development script with nodemon for better development experience
- Added basic error handling middleware
- Improved server startup configuration

## Type of Change

## Related Issues
Fixes #[issue-number] - Long deployment build times and server configuration optimization

## Testing Done
- Verified server starts correctly with `node server.js`
- Tested static file serving with caching headers
- Verified error handling middleware functionality 
- Tested development mode with nodemon
- Confirmed all routes are working as expected

## Testing Steps to Verify:
1. Run `npm install` to install dependencies
2. Start server with `node server.js` or `npm run dev`
3. Access http://localhost:3000 to verify static file serving
4. Verify server logs show proper startup message

## Checklist
- [x] All existing functionality continues to work as expected

## Additional Notes
- Package.json has been simplified to remove unused dependencies which should improve build times
- Added proper error handling which was missing in the original implementation
- Server configuration is now more production-ready with proper caching and error handling

## Deployment Notes
- Node.js version >= 14 is required
- Install dependencies with `npm install` before starting the server
- Basic server commands:
 - Production: `npm start`
 - Development: `npm run dev`
- Environment variables can be configured for different deployment environments

---
*Server optimization and configuration improvements for better performance and maintainability*
